Domain step test: Add the free domain for a year explainer textbox

niranjan-uma-shankar opened the pull request:
#### Changes proposed in this Pull Request

This PR builds on top of the domain-copy A/B test introduced in https://github.com/Automattic/wp-calypso/pull/37546. Check the `What next?` section in that PR on how to review this and other PRs as part of this A/B test.

* Add the free domain for a year explainer textbox below the domain input box. The textbox is shown only when there are domain search results displayed.
* The CTA on the textbox will take you to the plans page with the free site option hidden(as discussed in pbm1bU-2G#comment-67-p2)
* For detailed context, check p8Eqe3-Uq-p2.

<img width="1375" alt="Screenshot 2019-11-20 at 8 24 31 PM" src="https://user-images.githubusercontent.com/1269602/69250202-41512d00-0bd5-11ea-950f-bc678ca2b0ca.png">


#### Testing instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Go to /start and put yourself in the variant variantShowUpdates of the domainStepCopyUpdates A/B test.
* The free-domain explainer textbox should be displayed with the same format and spacing alignments as shown in the screenshot above.
* Click the `Review plan options ...` link and verify that the plan page on the next step _does not show_ the free site option

<img width="680" alt="Screenshot 2019-11-15 at 8 37 18 PM" src="https://user-images.githubusercontent.com/1269602/68954356-e04ae300-07e9-11ea-8055-a3897dccb7d0.png">

* Complete signup after clicking `Review plan options ...` and verify that a new site is created for you.


**Mobile View**

* Verify alignment and copy checks out in mobile views.

**Check in other domain pages**

* Login to any account, and after assigning yourself to the variant, check the launch flow for privates sites and the Add domain page. In both places, the domain step copy changes should not be seen(as we are showing the copy only during signup).

**Verify that the A/B test is not automatically assigned**
* Go to /start and without manually assigning yourself to an A/B test, just complete the signup flow. 
* In your Browser console, type `localstorage.ABTests;`. Verify that the `domainStepCopyUpdates ` test is not present in the object. This is because we do not yet want any user getting assigned to the test till all the pieces are deployed.

niranjan-uma-shankar commented on client/lib/signup/step-actions.js:
```diff
@@ -675,7 +683,7 @@ export function createUserAccountFromEmailAddress( callback, { email } ) {
  * Returns the dependencies for the step.
  *
- * @param {function} callback Callback function
+ * @param {Function} callback Callback function
```
Not capitalizing `f` gave lint errors.

niranjan-uma-shankar commented:
@Automattic/zelda This change touches some important signup files, so it'd be great to have your reviews.

victorespigares commented:
@niranjan-uma-shankar I'm noting that the space between the input and the explainer is not the same as in the Figma file. It looks more crowded in the PR:

This is the Figma:
![image](https://user-images.githubusercontent.com/538790/68960199-fba8f480-07cf-11ea-9ed0-76147e091b55.png)

Giving the search input a padding-top of 18px and a padding-bottom of at least 48px, it looks good in both steps.

I'll go now through the testing instructions.

niranjan-uma-shankar commented:
Thanks @victorespigares , I've pushed in a change with the suggested padding changes. Also, with all other changes for the domain step merged, I just rebased so this PR represents how the UI will look with all pieces together.

jancavan commented:
1. Can we possibly use whole numbers for the pricing? Including cents add a whole lot of complexity to the page. I think we are using such format to accommodate a variety of currencies, but IIRC, I think there's areas in product where we use whole numbers instead.

2. I realized this particular line may actually not work just in the current context of the page. Saying "next page" implies that it's the immediate next page, but the Plan lineup doesn't actually show up until they type in a domain.

My vote is to remove the entire line completely. This would help address the long copy.

![Screenshot 2019-11-18 10 22 15](https://user-images.githubusercontent.com/4924246/69079143-312b2900-09ee-11ea-80b4-be6d91771acf.png)

3. The text block has too much top/bottom padding. Can we reduce it like so?

![Screenshot 2019-11-18 10 40 11](https://user-images.githubusercontent.com/4924246/69079894-d1358200-09ef-11ea-8de7-1a3a616c47a8.png)

I took chopped off probably about -20px from the top and -30px from the bottom.

4. I agree with the copy being quite long. It looks particularly bad on mobile. We could remove the first paragraph (as mentioned in 2) and shorten the sub-text:

![Screenshot 2019-11-18 11 36 07](https://user-images.githubusercontent.com/4924246/69083860-d4347080-09f7-11ea-901f-b1eac700ad53.png)

The link can probably be omitted too; we can just have [ go with any paid plan ] link to the Plans page.

5. I made font size changes in: 623d2cc75729f744af2affa87d57c1358f85277a. I also removed some padding that appears on mobile, I don't think we need those.

@niranjan-uma-shankar  @victorespigares @donalirl

jancavan commented:
With this change: https://github.com/Automattic/wp-calypso/commit/623d2cc75729f744af2affa87d57c1358f85277a, I realized the text does need padding on mobile, but only when it's white on dark. I'll circle back to it later.

niranjan-uma-shankar commented:
> I realized this particular line may actually not work just in the current context of the page.

The copy is now updated

> The text block has too much top/bottom padding. Can we reduce it like so?

Padding changes are done.

> I realized the text does need padding on mobile, but only when it's white on dark.

I've updated the padding for mobile when it's white on dark.

niranjan-uma-shankar commented:
> This likely isn't blocking if we can make sure user is assigned to control group @apeatling

@adamziel is this something that is taken care of already, or should I be handling it in this PR? I'm guessing the way this should be done is to exclude the A/B test assignment on horizon environment. Or is there another way?

jancavan commented:
Made a small change in: https://github.com/Automattic/wp-calypso/pull/37554/commits/dc94346db69c4bcb5266e1143a24e20de20ef162 to reduce the top-margin of the recommended pills. @victorespigares @niranjan-uma-shankar 

I found this bug which I can only repro in this PR:

![Screenshot 2019-11-20 09 33 20](https://user-images.githubusercontent.com/4924246/69262990-47b1bb80-0b79-11ea-8c28-64538e04710a.png)

Steps I took:

1. Picked `Business` vertical
2. Picked `Food`
3. Left "Business name" empty
4. Picked "Free" from the Search results (but from testing, it also appears if you pick a paid domain)
5. Notice both "Your Plan" and "Popular" badges appear on the plan lineup

niranjan-uma-shankar commented:
@jancavan Were you using the "Add new site" option? It's a known bug that using "Add new site" shows the Your Plan label. More about it in the bug report here - https://github.com/Automattic/wp-calypso/issues/36712#issuecomment-544423842

The label "Your Plan" should not show if you are going through a fresh signup.

jancavan commented:
Ah, good to know, thanks @niranjan-uma-shankar.

andrewserong commented on client/components/domains/free-domain-explainer/index.jsx:
```diff
@@ -0,0 +1,52 @@
+		return (
+			<div className="free-domain-explainer card is-compact">
+				<header>
+					<h1 className="free-domain-explainer__title">
```
There's already an `<h1>` element on the page, so should this one be an `<h2>`?

andrewserong commented on client/components/domains/register-domain-step/index.jsx:
```diff
@@ -1167,12 +1177,10 @@ class RegisterDomainStep extends React.Component {
-		) {
-			return this.renderExampleSuggestions();
+		if ( suggestions.length === 0 && ! this.state.loadingResults ) {
+			if ( this.props.showExampleSuggestions ) {
```
Does this `if` statement change do anything different to the previous version?

niranjan-uma-shankar replied:
This just restores to the previous version and does nothing different. Reason for restoring this is that we decided to retain the domain example box.

andrewserong commented on client/lib/signup/step-actions.js:
```diff
@@ -174,6 +174,14 @@ export function createSiteWithCart( callback, dependencies, stepData, reduxStore
 			siteType ||
 			getSiteVertical( state );
 		newSiteParams.find_available_url = true;
+	} else if ( get( getSignupDependencyStore( state ), 'shouldHideFreePlan', false ) ) {
```
Looks like the `newSiteParams` here are the same as for the `isDomainStepSkippable` block above — could these be combined, or did you want to keep them separate in case there are differences between them in the future?

niranjan-uma-shankar replied:
Addressed in [00f8cc2](https://github.com/Automattic/wp-calypso/pull/37554/commits/00f8cc226c263c8b4f42037c189f453ee734c181)

andrewserong commented on client/signup/steps/domains/index.jsx:
```diff
@@ -213,24 +208,29 @@ class DomainsStep extends React.Component {
+				suggestion &&
+				( isPurchasingItem
+					? suggestion.domain_name
+					: suggestion.domain_name.replace( '.wordpress.com', '' ) ),
```
This is minor style nit, but this set of declarations is starting to get pretty long. What do you think about breaking this up into separate `const` declarations for each of these (`suggestion`, `isPurchasingItem`, `siteUrl`, and `domainItem`?

niranjan-uma-shankar replied:
Addressed in [00f8cc2](https://github.com/Automattic/wp-calypso/pull/37554/commits/00f8cc226c263c8b4f42037c189f453ee734c181)

niranjan-uma-shankar commented:
Thanks for the review!

> My only other thought is if it'd be good to add in translate calls for the text in the copy, in case this looks like text that you want to keep (or run outside of en locales) and to help prevent delaying translation further down the track, but again, not a blocker 

I'd prefer to wait till we are sure that the test wins. A one week delay in deploy to wait for translations should be fine I guess. However, if the test is unsuccessful, we would have incurred  translation costs that could've been avoided.

andrewserong commented:
> However, if the test is unsuccessful, we would have incurred translation costs that could've been avoided.

That’s a great point — good idea to leave it for now, then 🙂

andrewserong commented:
@victorespigares I've put together a small PR to fix the issue with the Plan cards and how they display on narrow screenwidth: https://github.com/Automattic/wp-calypso/pull/37813

klimeryk commented on client/components/domains/free-domain-explainer/index.jsx:
```diff
@@ -0,0 +1,50 @@
+							borderless
+							className="free-domain-explainer__subtitle-link"
+							onClick={ this.handleClick }
+							href
```
FYI, this line is causing a warning in the console:

```
backend.js:6 Warning: Failed prop type: Invalid prop `href` of type `boolean` supplied to `Button`, expected `string`.
```
That's because it's getting a `true` if you use that form. You should either explicitly assign an empty string value to it (`href={ '' }`) or better yet, just omit the attribute.

klimeryk replied:
Added a quickfix for it here: https://github.com/Automattic/wp-calypso/pull/39092/commits/04ab81642a17ddd3d55b51e57a07b4397f68caf4 (part of https://github.com/Automattic/wp-calypso/pull/39092) 🙇
